tests/README.rst

```diff
@@ -80,6 +80,125 @@ To do that, run `podman pull quay.io/rhinstaller/anaconda-ci:master` or build
+For external contributors, an organization member needs to approve the test run
+by sending a comment starting with ``/tests``.
+
+
+What role do the containers have
+--------------------------------
+
+All active branches run tests in containers. Containers have all the
+dependencies installed and the environment prepared to run tests or connect our
+GitHub runners (used by RHEL only).
+
+Automatic container build
+_________________________
+
+Containers are updated daily by `.github/workflows/container-autoupdate.yml`_
+GitHub action from Anaconda ``master`` repository. Before pushing a new
+container, tests are executed on this container to avoid regressions.
+
+Manual container build
+______________________
+
+There are two possible ways how to build containers manually.
+
+GitHub: Just go to the
+`actions tab <https://github.com/rhinstaller/anaconda/actions?query=workflow%3A%22Refresh+container+images%22>`_
+in the Anaconda repository to the “Build, test and push container“ and press the
+``Run workflow`` button on a button on a particular branch. Usually ``master``,
+but for testing a change to the container you can push your branch to the
+origin repo and run it from there.
+
+Security precautions for testing RHEL
+-------------------------------------
+
+Getting into our host/internal network
+______________________________________
+
+One of the main precautions is that each container test run has
+a limited time and is destroyed after timeout/end of test. That should narrow
+what attackers could do or how they can create a backdoor. See the image for
+more info:
+
+.. image:: docs/images/tests/GH-self-hosted-runners.png
+
+
+Another hardening of this is potential issue is that only PRs
+approved by/created by users with permission to write are able to run the tests.
+To achieve this we have two ways how to start the test.
+
+**PR created by rhinstaller member** -- these are started from the RHEL branch
+workflow file by ``pull_request_target`` as usual. This workflow has two
+dependent jobs. First will check user privileges, second will run the tests in
+case the first one succeeded.
+
+**PR created by external contributors** -- these have to be started by workflow
+file `.github/workflows/validate-rhel-8.yml`_ from the ``master`` branch
+checking all the comments. If comment starts with ``/test`` phrase it will check
+the owner of the comment. When everything succeed it will set progress on the pull
+request originating the comment and start the tests. This progress is updated
+based on the result of the tests. As explained above, the whole implementation
+of the workflow is in the ``master`` branch which could be pretty confusing.
+
+Changing workflow file by attacker
+__________________________________
+
+Because test description is part of the repository attackers may change
+workflow files by creating PR to do their malicious attack. This could normally
+happen. Because of that we are using ``pull_request_target`` instead of
```

Comment:
"This could normally happen", I'd drop this. The next sentence explains this in better detail.